### ISSUE TITLE
CA2252 false positive

### ISSUE THREAD

**Issue by meziantou** [dotnet/roslyn-analyzers#5366]
### Analyzer

**Diagnostic ID**: CA2252: `'A' implements a preview interface. See https://aka.ms/dotnet-warnings/preview-features for more information.`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: SDK 6.0.100-preview.7.21379.14

### Steps To Reproduce

````
public class A : IClassFixture<A> // CA2252
{
}

public interface IClassFixture<TFixture> where TFixture : class
{
}
````

````
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>net6.0</TargetFramework>
    <LangVersion>10.0</LangVersion>
    
    <AnalysisLevel>latest</AnalysisLevel>
    <EnableNETAnalyzers>true</EnableNETAnalyzers>
    <AnalysisMode>AllEnabledByDefault</AnalysisMode>
  </PropertyGroup>
</Project>
````

### Expected behavior

- No diagnostic reported as the interface is not decorated with the Preview attribute
- The link does redirect to a documentation page

### Actual behavior

- A diagnostic is reported
- The link redirect to a bing search

**Comment by RikkiGibson** [comment]
Should the analyzer check that the interface was declared in an assembly with a particular name or simple name?

**Comment by McNerdius** [comment]
I'm seeing this on 6.0.100-rc.1.21410.68:

`dotnet new classlib`, no changes to the `.csproj`

Class1.cs:

```
interface foo
{
}

interface bar : foo // CA2252
{
}
```

If i add a property to `foo` the warning goes away.

**Comment by huoyaoyuan** [comment]
Also observing this when implementing an empty interface.

**Comment by pgovind** [comment]
> Version: SDK 6.0.100-preview.7.21379.14
> dotnet new classlib, no changes to the .csproj

Yea we were still finalizing the implementation in P7 (and late into RC1) :) At that point, there was some thought that empty interfaces should trigger a preview diagnostic. With the latest changes in the release/6.0.1xx branch, both examples don't reproduce and work as expected. Closing the issue. Thanks for reports everyone!! Much appreciated!
